### PR TITLE
Idemia Microservice Database Update for UEID and CSP Custom ID entries

### PR DESCRIPTION
**Pull request by pshumateEDC**
UEID and Kong managed CSP Custom IDs added to enrollment record table in Idemia Microservice DB. Primary key set to autogenerated value. Record lookup still set to `record_csp_uuid` but no longer as primary key. CSP Custom ID is scraped from request header. In order to ensure that records are unique across multiple CSPs, the `record_csp_uuid` and `record_csp_id` are unique together. This allows different CSPs to use the same UUID when enrolling an individual without causing a collision.

### CODE REVIEW

File: api/tests/test_enrollment_records.py [hunk 1]
```diff
@@ -6,22 +6,33 @@
...
-    return {"record_uuid": uuid.uuid4(), "first_name": "Bob", "last_name": "Testington"}
+    return {
+        "record_csp_uuid": uuid.uuid4(),
+        "first_name": "Bob",
+        "last_name": "Testington",
+    }
 
 
-def create_enrollment_record(client):
+def generate_meta_data(consumer_id) -> dict:
```
Comment (folksgl) [inline api/tests/test_enrollment_records.py:21]: Probably better in this instance to make `consumer_id` default to `consumer_a` and just not provide the parameter unless you explicitly want to test something different.

```
def generate_meta_data(consumer_id="consumer_a") -> dict:
```

File: api/tests/test_enrollment_records.py [hunk 2]
```diff
@@ -6,22 +6,33 @@
...
+    """ Helper method for generating headers for all requests to the microservice """
+    return {"HTTP_X_CONSUMER_CUSTOM_ID": consumer_id}
+
+
+def create_enrollment_record(client, consumer_id):
```
Comment (folksgl) [inline api/tests/test_enrollment_records.py:26]: Probably better in this instance to make `consumer_id` default to `consumer_a` and just not provide the parameter unless you explicitly want to test something different.

```
def generate_meta_data(consumer_id="consumer_a") -> dict:
```

File: api/tests/test_enrollment_records.py [hunk 3]
```diff
@@ -6,22 +6,33 @@
...
     """
     Perform a POST operation to the enrollment endpoint. This operation will
     serve as the base for many other tests, and will intentionally generate
     failures if the operation results in anything other than a 201 response
     """
     url = reverse("enrollment")
     record_data = generate_enrollment_record_data()
-    response = client.post(url, record_data)
+    meta_data = generate_meta_data(consumer_id)
+    response = client.post(url, record_data, **meta_data)
```
Comment (folksgl) [inline api/tests/test_enrollment_records.py:35]: If the meta_data variable is supposed to be a dict of headers, then the request should probably assign the `headers` parameter in `request.post()`. See https://requests.readthedocs.io/en/latest/api/#requests.request for the full list.

To me, this also probably means that `meta_data` is in need of renaming to be more clear/accurate to its intent.

File: api/tests/test_enrollment_records.py (hunk 2 above)
Comment (pshumateEDC) [reply api/tests/test_enrollment_records.py]: Fixed and updated throughout all unit tests

File: api/tests/test_enrollment_records.py [hunk 4]
```diff
@@ -6,22 +6,33 @@
...
+    return {
+        "record_csp_uuid": uuid.uuid4(),
+        "first_name": "Bob",
+        "last_name": "Testington",
+    }
 
 
-def create_enrollment_record(client):
+def generate_meta_data(consumer_id) -> dict:
+    """ Helper method for generating headers for all requests to the microservice """
```
Comment (folksgl) [inline api/tests/test_enrollment_records.py:22]: If the purpose of this method is to generate a dict of headers for a request object, it probably shouldn't obfuscate that with terms like `meta_data` and should use standard `requests` library terminology.

File: api/tests/test_enrollment_records.py (hunk 3 above)
Comment (pshumateEDC) [reply api/tests/test_enrollment_records.py:35]: Is the solution just a rename? All requests require this header.

File: api/tests/test_enrollment_records.py [hunk 5]
```diff
@@ -6,22 +6,33 @@
...
+    return {
+        "record_csp_uuid": uuid.uuid4(),
+        "first_name": "Bob",
+        "last_name": "Testington",
+    }
 
 
-def create_enrollment_record(client):
+def generate_meta_data(consumer_id="consumerA") -> dict:
+    """ Helper method for generating headers for all requests to the microservice """
```
Comment (folksgl) [inline api/tests/test_enrollment_records.py:22]: The docstring can probably be shortened to
```
""" Helper method for generating enrollment record request headers """
```

Comment (pshumateEDC) [reply api/tests/test_enrollment_records.py]: Comment updated.

File: api/tests/test_enrollment_records.py (hunk 4 above)
Comment (pshumateEDC) [reply api/tests/test_enrollment_records.py]: Updated method name for clarity.

**Pull request merged**